lib/model.js

```diff
@@ -842,7 +842,19 @@ class Model {
       }
     });
 
-    this.options.indexes = this.options.indexes.map(this._conformIndex);
+    this.options.indexes = this.options.indexes.map(index => {
+      index = this._conformIndex(index);
+      //Add unique indexes to rawAttributes
+      if (index.unique && index.fields) {
+        index.fields.forEach(field => {
```

Comment:
@sushantdhiman Composite index also works. I have added an unit test for it but I'll add an integration test also.